gcsio/src/main/java/com/google/cloud/hadoop/gcsio/authorization/StorageRequestAuthorizer.java

```
@@ -0,0 +1,139 @@
+            createGCSURI(bucketName, ((Storage.Objects.Get) request).getObject()));
+      } else if (request instanceof Storage.Objects.Delete) {
+        authorizationHandler.handleDeleteObject(
+            createGCSURI(bucketName, ((Storage.Objects.Delete) request).getObject()));
+      } else if (request instanceof Storage.Objects.Rewrite) {
+        authorizationHandler.handleRewriteObject(
+            createGCSURI(
+                ((Storage.Objects.Rewrite) request).getSourceBucket(),
+                ((Storage.Objects.Rewrite) request).getSourceObject()),
+            createGCSURI(
+                ((Storage.Objects.Rewrite) request).getDestinationBucket(),
+                ((Storage.Objects.Rewrite) request).getDestinationObject()));
+
+      } else if (request instanceof Storage.Objects.Copy) {
+        authorizationHandler.handleCopyObject(
+            createGCSURI(
+                ((Storage.Objects.Copy) request).getSourceBucket(),
+                ((Storage.Objects.Copy) request).getSourceObject()),
+            createGCSURI(
+                ((Storage.Objects.Copy) request).getDestinationBucket(),
+                ((Storage.Objects.Copy) request).getDestinationObject()));
+      } else if (request instanceof Storage.Objects.Patch) {
+        authorizationHandler.handlePatchObject(
+            createGCSURI(bucketName, ((Storage.Objects.Patch) request).getObject()));
+      }
+
+      // Bucket
+      else if (request instanceof Storage.Buckets.List) {
+        authorizationHandler.handleListBucket();
+      } else if (request instanceof Storage.Buckets.Insert) {
+        authorizationHandler.handleInsertBucket(createGCSURI(bucketName, null));
+      } else if (request instanceof Storage.Buckets.Get) {
+        authorizationHandler.handleGetBucket(createGCSURI(bucketName, null));
+      } else if (request instanceof Storage.Buckets.Delete) {
+        authorizationHandler.handleDeleteBucket(createGCSURI(bucketName, null));
+      }
+  }
```

It seems that we need to fail if this is unknown method, otherwise it can become a security hole.